src/main/java/com/epam/reportportal/utils/properties/SystemAttributesExtractor.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2019 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.epam.reportportal.utils.properties;
+
+import com.epam.ta.reportportal.ws.model.attribute.ItemAttributesRQ;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import static java.util.Optional.ofNullable;
+
+/**
+ * @author <a href="mailto:ivan_budayeu@epam.com">Ivan Budayeu</a>
+ */
+public class SystemAttributesExtractor {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(SystemAttributesExtractor.class);
+
+	private static final String ATTRIBUTE_VALUE_SEPARATOR = "|";
+
+	private SystemAttributesExtractor() {
+		//static only
+	}
+
+	public static Set<ItemAttributesRQ> extract(final String resource) {
+		Properties properties = new Properties();
```

Comment:
Can you explain in few words how do these system attributes work?

Reply:
There are 2 types:
1) from a `property` file - for now it's only `agent name` and `agent version`
2) from the `System` - OS name, arch and version 
All key/value are parsed from the properties and converted to `ItemAttributesRQ` with `system` field equal to `true`